packages/core/src/TagsInput/TagsInput.js

```diff
@@ -163,16 +167,17 @@ const HvTagsInput = (props) => {
     (event, tag) => {
       event.preventDefault();
       if (tag !== "") {
-        const newTagsArr = [...value, { label: tag, type: "semantic" }];
+        const newTag = { label: tag, type: "semantic" };
+        const newTagsArr = [...value, newTag];
         setValue(newTagsArr);
         setTagInput("");
         setTagCursorPos(newTagsArr.length);
```

Comment:
The thing here is that this will clear the input box even if the controlled HvTagsInput doesn't get updated. Imagine I receive an onChange/onAdd/whatever and decide not to update the value, but instead display an error... The text the user wrote is already lost.

I would move these two line to an useEffect, triggered when the value changes.